File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    tmp = []
    for progress, speed in zip(progresses, speeds):
        cnt = 0
        while progress < 100:
            progress += speed
            cnt += 1
        tmp.append(cnt)
    length = len(tmp)
    cnt = 1
    idx = 0
    for i in range(1, length):
        check = tmp[idx]
        if(tmp[i] > check):
            answer.append(cnt)
            idx = i
            cnt = 1
        else:
            cnt += 1
        if(i == length - 1):
            answer.append(cnt)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    

    public int[] solution(int[] progresses, int[] speeds) {
        int[] answer;
        List<Integer> ans = new ArrayList<>();
        int length = progresses.length;
        int idx = 0;
        int tmp = 0;
        while (true) {
            if (idx == length) {
                break;
            }
            for (int i = idx; i < length; i++) {
                progresses[i] += speeds[i];
            }
            tmp = 0;
            while (true) {
                if (progresses[idx] >= 100) {
                    tmp += 1;
                    idx += 1;
                    if (idx == length) {
                        break;
                    }
                } else {
                    break;
                }
            }
            if (tmp != 0) {
                ans.add(tmp);
            }
        }
        answer = ans.stream().mapToInt(Integer::intValue).toArray();
        return answer;
    }
}